### PR TITLE
Remove old & stale bpf maps

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

File: daemon/daemon/daemon.go [hunk 1]
```diff
@@ -222,6 +225,10 @@ func (d *Daemon) init() error {
 
 // NewDaemon creates and returns a new Daemon with the parameters set in c.
 func NewDaemon(c *Config) (*Daemon, error) {
+	if Instance != nil {
```
Comment (aanm) [inline daemon/daemon/daemon.go]:
``` go
import "sync"
var(
    once sync.Once
)

// NewDaemon creates and returns a new Daemon with the parameters set in c.
func NewDaemon(c *Config) (d *Daemon, err error) {
    once.Do(func() {
        d, err = newDaemon(c)
    })
    return nil, fmt.Errorf("Daemon instance already created")
}

func newDaemon(c *Config) (*Daemon, error){
    if c == nil {
        return nil, fmt.Errorf("Configuration is nil")
    }
....
```

File: daemon/daemon/daemon.go [hunk 2]
```diff
@@ -338,7 +347,43 @@ func NewDaemon(c *Config) (*Daemon, error) {
...
-	return &d, nil
+	if err := filepath.Walk(common.BPFCiliumMaps, staleMapWalker); err != nil {
+		log.Warningf("Error while scanning for stale maps: %s", err)
+	}
+
+	return d, nil
+}
+
+func checkStaleMap(path string, filename string, id string) {
+	if tmp, err := strconv.ParseUint(id, 0, 16); err == nil {
```
Comment (aanm) [inline daemon/daemon/daemon.go]: don't miss the `d.endpointsMU.Lock()` and `d.endpointsMU.Unlock()`

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/daemon/daemon.go (hunk 2 above)
Comment (tgraf) [reply daemon/daemon/daemon.go]: Can something already happen in parallel at this point?

Comment (aanm) [reply daemon/daemon/daemon.go]: @tgraf Yes, `SyncState` was called previously.

``` go
// SyncState syncs cilium state against the containers running in the host. dir is the
// cilium's running directory. If clean is set, the endpoints that don't have its
// container in running state are deleted.
func (d *Daemon) SyncState(dir string, clean bool) error {
    restored := 0

    log.Info("Recovering old running endpoints...")

    d.endpointsMU.Lock()
...
```

**Comment by tgraf** [comment]
@aanm I've reverted the Daemon singleton. Fixing up the test cases would take a while

**Review by aanm — APPROVED** [review]

**Pull request merged**